### ISSUE TITLE
CC0014 should place the three elements of the ternary on separat lines if the total length would exceed 72 characters.

### ISSUE THREAD

**Issue by kwhitefoot** [code-cracker/code-cracker#743]
#Bug

````vb.net
         if veryLongCondition(withALongArgumentList) Then
             aVariable = aLongfunctionName(withAnotherLongArgList)
         else
             aVariable = anEvenLongerfunctionName(withYetAnotherLongArgList)
         end if
````

Current output after fix applied (if it is a code fix bug):

````vb.net         
    aVariable = if(veryLongCondition(withALongArgumentList), aLongfunctionName(withAnotherLongArgList), anEvenLongerfunctionName(withYetAnotherLongArgList))
````

Expected output after fix applied (if it is a code fix bug):

````vb.net
    aVariable = if(veryLongCondition(withALongArgumentList),
                          aLongfunctionName(withAnotherLongArgList), 
                          anEvenLongerfunctionName(withYetAnotherLongArgList))
````

**Comment by giggio** [comment]
I don't think this is a bug. I will leave it as an enhancement for this code fix.
The problem is that not everyone breaks on 72 characters. We need to take some information from the options when applying the fix. Is there such a configuration in Visual Studio, VB or C#?

**Comment by kwhitefoot** [comment]
Oops, yes of course this is not a bug.  I suppose I should have added a comment to the same effect elsewhere.  True enough not everyone breaks the lines at column 72 (or any other column) but I think most would agree that it is hard to read diffs when the lines are long and simply putting each element of the ternary on a separate line seems to me a reasonable compromise.

And, no, as far s I can tell VS does not offer this kind of formatting option.  I display a vertical line at 72 characters so that I can see where to manually break the lines and then manually indent to line up brackets, etc..  The VS editor is remarkably weak in this area compared with, for instance, Emacs.  I suppose you could have a suggested add-on that provides the column number.  I use Editor Guidelines, https://visualstudiogallery.msdn.microsoft.com/da227a0b-0e31-4a11-8f6b-3a149cf2e459, but I don't know where it keeps its settings.

My reason for wanting it is that I deal with a lot of legacy code written by people who didn't bother breaking lines at all so I can easily end up with 500 character lines when using this fix.

**Comment by giggio** [comment]
I could not find an option setting that would define the maximum length of a line before a line break.
The line breaking option could be applied to other code fixes as well. This is not the only one.
@code-cracker/owners what do you think? The problem here is that we would have to always leave this on, or always off (which is what we have right now).
@kwhitefoot I don't think everyone agrees on this issue. I, myself, very often, write long lines.

**Comment by kwhitefoot** [comment]
@giggio If you write very long lines how do you make sense of diffs?

Another argument for placing the elements on separate lines is that it preserves the visual clues that make it easy to see that this is a conditional construct.  In fact I would argue that putting them on separate lines is less controversial than making one long line because they were all on separate lines to begin with.

Disagreeing is good.  If we all agreed all the time we would go full speed in the wrong direction far too often. :-)

**Comment by jwooley** [comment]
I wonder if a better option for the long line issue would be having a refactoring which takes a method with multiple parameters on a single line and breaks them up on multiple lines adding named parameters. It wouldn't necessarily help the case of ternary operators, but might be useful as a separate tool.

**Comment by kwhitefoot** [comment]
@jwooley Sounds like a good idea to me, perhaps you could suggest it formally.

Perhaps it should offer multiple lines and named parameters as separate fixes so that you can have one or the other or both.  Not everyone likes named parameters but I think they help a lot especially when you have a long argument list with a lot or arguments of the same type..

**Comment by giggio** [comment]
I like @jwooley'd idea. Maybe offer fixes for breaking at 80, 72, 120 columns...
@kwhitefoot I know it is not great for diffs, but very often I work on really large monitors and then I actually don't notice. I am on VS right now with 180 columns on a 3 monitors setup...

**Comment by kwhitefoot** [comment]
How about offering two or more fixes for the same diagnostic:

1. All on one line
2. On three lines

And perhaps:
3. Every parameter on a separate line
4. As 3. but with named parameters

**Comment by kwhitefoot** [comment]
@giggio I too have quite a lot screen area (two side by side 60cm/24" monitors) but long lines are still a pain to read.
